### PR TITLE
Wrap the code in a package

### PR DESCRIPTION
**Pull request by kiwiroy**
This pull request wraps the function in a package. Mainly using `usethis` defaults and moving the documentation you already had into `roxygen` blocks.

I've added some test data that quickly enabled me to test out this, hopefully it will help others too. The readme is updated to reflect this. Another benefit is also testing the code in the `testthat` test suite, although. quite minimally currently.

Thanks for the codes and considering this.

### CODE REVIEW

**Comment by JimWhiting91** [comment]
This is such a help, thanks for putting this all together. The addition of the example data is very useful and looking at the edits you've made to how tmp files are being handled are definitely a more sophisticated approach. I'll try and take a look at all the edits in full this weekend, I'm not so familiar with a few of the changes and I just want to make sure I understand your edits before merging. Thanks again!

**Comment by kiwiroy** [comment]
Fully understand and I'm happy to address any edits required. [R Packages](https://r-pkgs.org/index.html) is a good resource if you're in need.

File: R/genotype_plot.R
Comment (JimWhiting91) [inline R/genotype_plot.R]: split here reorders populations so they're alphabetical in popmap2 as opposed to the order they appear in the unique(popmap[,2]), which is the original functionality. Would just need to add an additional line to reorder popmap2 so names(popmap2) == unique(popmap[,2]), or revert to the original unless there is another reason to make this change?

**Comment by JimWhiting91** [comment]
Aside from that comment I've been through the other files and am happy to merge once the popmap2 issue is addressed, thanks again. Also thanks for sending the package documentation through as well, I have another package that I'll be looking to upload soon so I'll be sure to have all of this set up beforehand next time.

**Review by JimWhiting91 — APPROVED** [review]
I've just gone in and reverted back to the original as I don't think there's any functionality difference and we retain the user-defined popmap order

**Comment by JimWhiting91** [comment]
I noticed after merging these at the example data contains multiallelic sites that (at the moment) the plotting function doesn't like. I've updated the example data with multiallelic and indels removed, and have also make a couple of changes to the readme that I should've clocked first time around. This is my first use of github from a developing perspective so still figuring things out a bit! Thanks again for your help Roy

**Comment by kiwiroy** [comment]
@JimWhiting91 all good, sorry I didn't help more, time ran away and I didn't commit the changes below to solve the reordering. It creates new `data.table` rather than using as is, which has the benefit of setting `colnames` to know internal ones as well as keeping `names(popmap2$pop) == unique(popmap[,2])`.

```patch
diff --git a/R/genotype_plot.R b/R/genotype_plot.R
index 9e1b854..0b5a0d7 100644
--- a/R/genotype_plot.R
+++ b/R/genotype_plot.R
@@ -40,14 +40,14 @@ genotype_plot<-function(vcf=NULL,
   }
   
   # Tidy up the popmap and turn into a list
-  popmap <- as.data.table(popmap)
-  popmap2 <- lapply(split(popmap[,1], popmap[,2]), unlist, use.names = FALSE)
+  popmap  <- data.table(ind = as.character(popmap[,1]), pop = popmap[,2])
+  popmap2 <- lapply(split(popmap, by = 'pop', keep.by = FALSE), unlist, use.names = FALSE)
   
   # Read in the subsetted VCF
   vcf_in<-read.vcfR(vcf_tempfile)
   
   # Filter the VCF for individuals...
-  vcf_in <- vcf_in[,c("FORMAT",unlist(popmap[,1]))]
+  vcf_in <- vcf_in[,c("FORMAT", popmap$ind)]
   
   # Remove invariants following filtering
   vcf_in <- vcf_in[is.polymorphic(vcf_in, na.omit=TRUE),]
```

P.S. Say hi to Prof J Bryant for me, if you see him.

**Pull request merged**
